server/src/main/java/org/opensearch/action/admin/indices/tiering/TieringUtils.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.indices.tiering;
+
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.cluster.routing.ShardRouting;
+import org.opensearch.cluster.routing.allocation.RoutingAllocation;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.common.util.FeatureFlags;
+import org.opensearch.index.IndexModule;
+
+/**
+ * Utility class for tiering operations
+ *
+ * @opensearch.internal
+ */
+@ExperimentalApi
+public class TieringUtils {
+
+    /**
+     *  Checks if the specified shard is a partial shard by
+     *  checking the INDEX_STORE_LOCALITY_SETTING for its index.
+     *  see {@link #isPartialIndex(IndexMetadata)}
+     * @param shard ShardRouting object representing the shard
+     * @param allocation RoutingAllocation object representing the allocation
+     * @return true if the shard is a partial shard, false otherwise
+     */
+    public static boolean isPartialShard(ShardRouting shard, RoutingAllocation allocation) {
+        IndexMetadata indexMetadata = allocation.metadata().getIndexSafe(shard.index());
+        return isPartialIndex(indexMetadata);
+    }
+
+    /**
+     * Checks if the specified index is a partial index by
+     * checking the INDEX_STORE_LOCALITY_SETTING for the index.
+     *
+     * @param indexMetadata the metadata of the index
+     * @return true if the index is a partial index, false otherwise
+     */
+    public static boolean isPartialIndex(final IndexMetadata indexMetadata) {
+        return FeatureFlags.isEnabled(FeatureFlags.TIERED_REMOTE_INDEX)
```

Comment:
IMO the feature flag check should be in where this is called from instead of in the Util class.

Reply:
makes sense, will update this

Reply:
i tried this, however this is leading to complicated if else logic at multiple places, so keeping it as is for now.

Reply:
It should just be an additional `&&` or `||` statement in the places that you're changing below anyways right? My concern here is that it makes the logic in `LocalShardsBalancer` not super clear without knowing that this feature flag exists in the util class. @harishbhakuni do you have a second opinion on this?

Reply:
+1 to this, I would also prefer using this feature flag logic directly in the `LocalShardsBalancer` for better clarity.

Reply:
yeah it made sense to me as well, but the logic was still not super readable. 
I have now moved the feature flag out of the function and added some comments to help make it more readable.